ejb3/src/main/java/org/jboss/as/ejb3/remote/ModuleAvailabilityRegistrarService.java

```diff
@@ -0,0 +1,357 @@
+ * as well as remote deployment events (provided by a distributed ServiceProviderRegistry) and communicate that
+ * module deployment information to local listeners (via ModuleAvailabilityRegistrarListemer).
+ * <p>
+ * This class is also suspend and resume aware, so that locally depployed modules will be marked as unavailable
+ * when the server is suspended, and marked as available when the server is resumed.
+ *
+ * As this service is only required to support EJB deployments, it should ideally be started ON_DEMAND and
+ * made available only when one or more EJBs are deployed on the server.
+ */
+public class ModuleAvailabilityRegistrarService implements ModuleAvailabilityRegistrar, Service {
+    private final ServiceDependency<SuspendableActivityRegistry> registryDependency;
+    private final ServiceDependency<ServiceProviderRegistrar<Object, GroupMember>> registrarDependency;
+    private final ServiceDependency<DeploymentRepository> repositoryDependency;
+
+    private SuspendableActivityRegistry registry;
+    private ServiceProviderRegistrar<Object, GroupMember> registrar;
+    private DeploymentRepository deploymentRepository;
+
+    private final DeploymentRepositoryListener deploymentRepositoryListener ;
+    private final SuspendableActivity activity;
+
+    private final List<ModuleAvailabilityRegistrarListener> listeners = new ArrayList<>();
+    private final Map<DeploymentModuleIdentifier, ServiceProviderRegistration<Object, GroupMember>> registrations = new HashMap<>();
+
+    private boolean started;
+
+    /**
+     * Create an instance of a ModuleAvailabilityRegistrar which reflects the content of a given DepoymentRepository.
+     *
+     * @param deploymentRepository the repository this registrar listens to for module availability information on a given host
+     * @param registrar            the ServiceProviderRegistrar instance used to store module availability information
+     * @param registry             the suspendable activity registry to register our server activity with
+     */
+    public ModuleAvailabilityRegistrarService(ServiceDependency<SuspendableActivityRegistry> registryDependency, ServiceDependency<ServiceProviderRegistrar<Object, GroupMember>> registrarDependency, ServiceDependency<DeploymentRepository> repositoryDependency) {
+        this.registryDependency = registryDependency;
+        this.registrarDependency = registrarDependency;
+        this.repositoryDependency = repositoryDependency;
+
+        this.activity = new ModuleAvailabilityRegistrarSuspendableActivity();
+        this.deploymentRepositoryListener = new ModuleAvailabilityRegistrarDeploymentRepositoryListener();
+
+        this.started = false;
+    }
+
+    // service interface
+    @Override
+    public boolean isStarted() {
+        return started;
+    }
+
+    @Override
+    public void start() {
+        EjbLogger.ROOT_LOGGER.info("Starting ModuleAvailabilityRegistrarService");
```

Comment:
If this log message is not temporary, this will have to be i18n-ed if it's meant to be on the info level. Or move it to lower level.